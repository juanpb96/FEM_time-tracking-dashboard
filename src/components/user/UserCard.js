import './user.scss';

export const UserCard = ({handleTimeframeChange}) => {
    return (
        <div className="user">
            <div className="user__info">
                <img 
                    className="user__picture" 
                    src="/assets/image-jeremy.png" 
                    alt="Jeremy's profile"
                />
                <h3>Report for</h3>
                <h2 className="user__name"><span>Jeremy</span> <span>Robson</span></h2>
            </div>
            <div className="user__timeframes">
                <button type="button" onClick={ () => handleTimeframeChange('daily') }>Daily</button>
                <button type="button" onClick={ () => handleTimeframeChange('weekly') }>Weekly</button>
                <button type="button" onClick={ () => handleTimeframeChange('monthly') }>Monthly</button>
            </div>
        </div>
    )
}
