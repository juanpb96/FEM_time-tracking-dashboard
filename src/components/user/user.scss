@import '/public/variables.scss';

.user {
    background-color: $dark-blue;
    border-radius: $border-radius;

    &__content {
        display: flex;
        align-items: center;
        gap: 20px;
        padding: 2rem 2.125rem;
        background-color: $blue;
        border-radius: inherit;
    }

    &__picture {
        width: calc(64px - 3px * 2);
        border: 3px solid #FFF;
        border-radius: 50%;
    }

    &__title,
    &__name {
        margin: 0;
    }

    &__title {
        margin-bottom: 0.25rem;
        color: $pale-blue;
        font: $fw-regular 0.9375rem/1.1113rem $ff-rubik;
    }

    &__name {
        color: #FFF;
        font: $fw-light clamp(1.5rem, 3vw, 2.5rem)/1.7775rem $ff-rubik;
    }

    &__timeframes {
        display: flex;
        padding: 1.5rem 0;

        button {
            flex: 1;
            color: $desaturated-blue;
            font: $fw-regular 1.125rem/1.3331rem $ff-rubik;
            background-color: transparent;
            border: none;
            cursor: pointer;
            transition: color .3s ease;
            
            &.user__active-timeframe {
                color: #FFF;
            }
        }
    }

}

@media only screen and (min-width: 768px) {
    .user {
        max-height: 32.375rem;
        min-width: 15.9375rem;

        &__content {
            padding: 2.375rem 2rem;
            flex-direction: column;
            align-items: flex-start;
            gap: 2.625rem;
        }

        &__picture {
            width: calc(78px - 3px * 2);
        }

        &__timeframes {
            padding: 1.625rem 2rem 2rem;
            flex-direction: column;
            align-items: flex-start;
            gap: 1.3125rem;
        }
    }
}

@media only screen and (min-width: 1000px) {
    .user {
        &__content {
            padding-bottom: 5rem;
        }

        &__name {
            line-height: 2.9625rem;
        }

        &__timeframes {
            button {
                &:hover {
                    color: #FFF;
                }
            }
        }
    }
}