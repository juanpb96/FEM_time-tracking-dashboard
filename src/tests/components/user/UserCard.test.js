import { shallow } from "enzyme"
import { UserCard } from "../../../components/user/UserCard";

describe('Tests in <UserCard />', () => {

    const handleTimeframeChange = jest.fn();

    test('should render the component correctly', () => { 
        
        const timeframe = 'daily';
        const wrapper = shallow(
            <UserCard 
                timeframe={ timeframe }
                handleTimeframeChange={ handleTimeframeChange }
            />
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('button.user__active-timeframe').text().trim().toLowerCase() ).toBe( timeframe );

    })

    test('should change selected timeframe to Monthly', () => { 

        const timeframe = 'daily';
        const wrapper = shallow(
            <UserCard 
                timeframe={ timeframe }
                handleTimeframeChange={ handleTimeframeChange }
            />
        );

        const expectedTimeframe = 'monthly';
        
        wrapper.find('button').get(2).props.onClick();
        expect( handleTimeframeChange ).toHaveBeenCalledWith( expectedTimeframe );

    })

})