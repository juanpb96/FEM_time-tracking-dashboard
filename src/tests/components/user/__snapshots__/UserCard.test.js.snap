// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <UserCard /> should render the component correctly 1`] = `
<div
  className="user"
>
  <div
    className="user__content"
  >
    <img
      alt="Jeremy's profile"
      className="user__picture"
      src="./assets/image-jeremy.png"
    />
    <div
      className="user__info"
    >
      <h3
        className="user__title"
      >
        Report for
      </h3>
      <h2
        className="user__name"
      >
        Jeremy Robson
      </h2>
    </div>
  </div>
  <div
    className="user__timeframes"
  >
    <button
      className="user__active-timeframe"
      onClick={[Function]}
      type="button"
    >
      Daily
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Weekly
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Monthly
    </button>
  </div>
</div>
`;
